Controllers/ReportsController.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using API.Models;
+using Microsoft.AspNetCore.Mvc;
+using API.Models.DTO.Administrator;
+using Microsoft.AspNetCore.Identity;
+using System.Collections.Generic;
+using System.Collections;
+using API.Models.DTO;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.AspNetCore.Authorization;
+using System;
+
+namespace API.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ReportsController : ApiControllerBase
+    {
+        public ReportsController(ApiContext context, UserManager<Employee> userManager) :
+        base(context, userManager){ }
+
+        [Authorize(Roles = "Admin")]
+        [HttpPost("report")]
+        public async Task<ActionResult<IEnumerable<GetDataDTO<PharmacyReportDTO>>>> GetPharmacyReport(FilterPharmacyReportDTO model)
+        {
+            if (!IsValidApiRequest())
+            {
+                return ApiBadRequest("Invalid Headers!");
+            }
+
+            var pharmacies = await Context.Report
+                    .Where(r => r.Pharmacy.Id == model.PharmacyId
+                    && DateTime.Compare(r.DateFrom, model.DateFrom) >= 0
+                    && DateTime.Compare(r.DateTo, model.DateTo) <= 0)
+                    .Select(r => new ReportDTO(
+                            r,
+                            Context.Employees.FirstOrDefault(e => e.Id == r.Employee.Id)
+                            .LastName))
+                            .ToListAsync();
+
+            var report = new PharmacyReportDTO(pharmacies);
+            report.AllAmount = pharmacies.Sum(s => s.OrderAmount);
```

Comment:
Can we move this sum LINQ to the constructor?

Reply:
Changed